lib/graphql/validation/rules/provided_non_null_arguments.ex

```diff
@@ -0,0 +1,61 @@
+
+defmodule GraphQL.Validation.Rules.ProvidedNonNullArguments do
+
+  alias GraphQL.Lang.AST.{Visitor, TypeInfo}
+  alias GraphQL.Type
+
+  defstruct name: "ProvidedNonNullArguments"
+
+  defimpl Visitor do
+
+    def enter(_visitor, _node, accumulator) do
+      {:continue, accumulator}
+    end
+
+    def leave(_visitor, %{kind: :Field} = node, accumulator) do
```

Comment:
Pattern matching on the node type when dealing with multiple nodes (I also need to deal with directives). I will review the other validators and make sure I'm being consistent.